engines/twp/detection_tables.h

```diff
@@ -0,0 +1,43 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Twp {
+
+const PlainGameDescriptor twpGames[] = {
+	{ "twp", "Thimbleweed Park" },
+	{ 0, 0 }
+};
+
+const ADGameDescription gameDescriptions[] = {
+	{
+		"twp",
+		nullptr,
+		AD_ENTRY1("ThimbleweedPark.ggpack1", nullptr),
```

Comment:
Actually it should be compatible with the different `Thimbleweed Park` versions, I own the GOG and Epic ones but there are other versions with different MD5.

Reply:
Yep, those need to be added as separate detection entries. This way we can have some control over what exactly users plug into ScummVM (the aim is to avoid known pirated versions, and have a safeguard against using ScummVM as an attack vector for malicious software). It's quite common for a single game to have 5-10 or more entries, especially in the age of webstores and online updates.

Reply:
The general (albeit imperfect) workflow is to add as many detection entries as we can, and let users provide us with hashes for unknown versions they own.